### PR TITLE
Tic-Tac-Toe-Plus 

### PR DESCRIPTION
**Pull request by gk-patel**
Hi, 

I have create a new varaint for the classic tic-tac-toe. I hope you guys like and will add it to the main branch. I have also added test scripts from ai, board and game. 

Thanks,

### CODE REVIEW

**Comment by flamecoals** [comment]
Hey, thanks for this! I am traveling tomorrow, so I will try to review this Sunday. :)

File: src/games/tictactoeplus/instructions.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Tic-Tac-Toe-Plus (ttt+) is a nice varaint of the classis Tic-Tac-Toe. In ttt+, there is a 4x4 board (vs. 3x3 in classic) and the game is played by two players.
+
+Usually, the 4x4 varaint of the classic game is considered baised or leads to a draw. In order to over come this limitation, ttt+ introduces a wildcard symbol - namely the "plus" symbol, which represent both a circle and a cross. Each time a player takes a turn, a dice is rolled to decide it the wildcard is shown or not, wherein a wildcard will be shown if the dice-outcome evaluates to 6. And, for a given game, the wildcard (or the plus) can appear only upto a maximum of 3 times.
```
[the remark this answers is not in the document]
Comment (gk-patel) [reply src/games/tictactoeplus/instructions.md]: Yes you are right, will change it.

File: src/games/tictactoeplus/index.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    { mode: GameMode.OnlineFriend },
+    { mode: GameMode.LocalFriend },
+  ],
+  description: 'A new variant with a plus',
+  descriptionTag: `Play Tic-Tac-Toe-Plus for \
+ free online. You can either do a single-player game against the computer,\
+ a multi-player game against a friend online or share your device and play\
+ locally against a friend.`,
+  instructions: {
+    videoId: 'USEjXNCTvcc',
```
[the remark this answers is not in the document]
Comment (gk-patel) [reply src/games/tictactoeplus/index.ts]: Yes i know about this, I will upload a video in the new year and add a link

[the remark this answers is not in the document]
Comment (jasonharrison) [reply src/games/tictactoeplus/index.ts]: Hi, I did a brief search and couldn't find any videos on YouTube for TTT+, do you know of any?

**Pull request merged**